src/Markdown/Parser.elm

```diff
@@ -684,6 +684,34 @@ parseAllInlinesHelp state rawBlocks parsedBlocks =
         [] ->
             Ok parsedBlocks
 
+endWithOpenBlockOrParagraph : RawBlock -> Bool
+endWithOpenBlockOrParagraph block = 
+    case block of
+    OpenBlockOrParagraph (UnparsedInlines str) -> 
+        not (String.endsWith (Debug.log "str" str) "\n")
+
+    ParsedBlockQuote blocks ->
+        case blocks of
+            last :: _ ->
+                endWithOpenBlockOrParagraph last
+            _ -> False
+    OrderedListBlock _ _ _ _ blockslist _ ->
+        case blockslist of 
+            blocks :: _ ->
+                case blocks of
+                    last :: _ ->
+                        endWithOpenBlockOrParagraph last
+                    _ -> False
+            _ -> False
+    _  -> False
+
+    --| UnorderedListBlock Bool Int (List CloseListItem) OpenListItem
```

Comment:
Here are some cases might needed to add into that function. I might look at these cases after I get a closer look at the spec